src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -3978,6 +3978,62 @@ public DefaultInterpolatedStringHandler(int literalLength, int formattedCount)
             );
         }
 
+        [Fact, WorkItem(58346, "https://github.com/dotnet/roslyn/issues/58346")]
+        public void UserDefinedConversion_AsFromTypeOfConversion()
+        {
+            var code = @"
+struct S
+{
+    public static implicit operator S(CustomHandler c) => default;
+
+    static void M()
+    {
+        /*<bind>*/S s = $"""";/*<bind>*/
+    }
+}
+";
+
+            var handler = GetInterpolatedStringCustomHandlerType("CustomHandler", "struct", useBoolReturns: false);
+
+            var comp = CreateCompilation(new[] { code, handler });
+            comp.VerifyDiagnostics(
+                // (8,25): error CS0266: Cannot implicitly convert type 'string' to 'S'. An explicit conversion exists (are you missing a cast?)
```

Comment:
>An explicit conversion exists (are you missing a cast? [](http://example.com/codeflow?start=89&length=54)

Is this accurate? Would explicit cast to ```S``` actually work? Consider adding a unit-test for this scenario.